src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,489 @@
+            /// Flushes any buffered data to the underlying stream asynchronously.
+            /// </summary>
+            /// <param name="cancellationToken">A cancellation token to observe while waiting for the flush operation to complete.</param>
+            /// <returns>A task representing the asynchronous flush operation.</returns>
+            public override async Task FlushAsync()
+            {
+                await this.jsonWriter.FlushAsync();
+            }
+
+            /// <summary>
+            /// Disposes the object.
+            /// </summary>
+            /// <param name="disposing">True if called from Dispose; false if called form the finalizer.</param>
+            protected override void Dispose(bool disposing)
+            {
+                // If there are unprocessed chars, encode and write them as the final block.
+                ReadOnlySpan<char> charsNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(0, numOfCharsNotWrittenFromPreviousChunk);
+
+                if (!charsNotProcessedFromPreviousChunk.IsEmpty)
+                {
+                    if (!this.isWritingJson)
+                    {
+                        int firstIndexToEscape = this.jsonWriter.NeedsEscaping(charsNotProcessedFromPreviousChunk);
+                        this.WriteChunk(charsNotProcessedFromPreviousChunk, charsNotProcessedFromPreviousChunk.Length, firstIndexToEscape, true);
+                        this.numOfCharsNotWrittenFromPreviousChunk = 0;
+                    }
+                    else 
+                    {
+                        WriteChunkWithoutEscaping(charsNotProcessedFromPreviousChunk, charsNotProcessedFromPreviousChunk.Length, true);
+                    }
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<char>.Shared.Return(this.buffer);
+                }
+
+                this.Flush();
+                base.Dispose(disposing);
+            }
+
+            /// <summary>
+            /// Asynchronously disposes of the current object and performs cleanup operations.
+            /// </summary>
+            /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+            public override async ValueTask DisposeAsync()
+            {
+                // If there are unprocessed chars, encode and write them as the final block.
+                ReadOnlyMemory<char> charsNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(0, numOfCharsNotWrittenFromPreviousChunk);
+
+                if (!charsNotProcessedFromPreviousChunk.IsEmpty)
+                {
+                    if (!this.isWritingJson)
+                    {
+                        int firstIndexToEscape = this.jsonWriter.NeedsEscaping(charsNotProcessedFromPreviousChunk.Span);
+                        this.WriteChunk(charsNotProcessedFromPreviousChunk.Span, charsNotProcessedFromPreviousChunk.Length, firstIndexToEscape, true);

```

Comment:
Named parameters would have some benefit here especially for the last argument: `isFinalBlock: true`